api/client/src/main/java/org/projectnessie/client/NessieConfigConstants.java

```diff
@@ -156,6 +156,30 @@ public final class NessieConfigConstants {
   public static final String CONF_NESSIE_OAUTH2_REFRESH_SAFETY_WINDOW =
       "nessie.authentication.oauth2.refresh-safety-window";
 
+  public static final String DEFAULT_IDLE_INTERVAL = "PT30S";
+
+  /**
+   * Config property name ({@value #CONF_NESSIE_OAUTH2_IDLE_INTERVAL}) for the OAuth2 authentication
+   * provider. The maximum idle interval to use; if the OAuth2 provider is not used after this
+   * interval, no more token refreshes will occur, until the provider is used again. Optional,
+   * defaults to {@value #DEFAULT_IDLE_INTERVAL}. Must be a valid <a
+   * href="https://en.wikipedia.org/wiki/ISO_8601#Durations">ISO-8601 duration</a>.
+   */
+  public static final String CONF_NESSIE_OAUTH2_IDLE_INTERVAL =
+      "nessie.authentication.oauth2.idle-interval";
+
+  public static final String DEFAULT_KEEP_ALIVE_INTERVAL = "PT30S";
+
+  /**
+   * Config property name ({@value #CONF_NESSIE_OAUTH2_KEEP_ALIVE_INTERVAL}) for the OAuth2
+   * authentication provider. The keep alive interval to use; if the OAuth2 provider background
```

Comment:
I think the description should be different - as it's from a user's perspective.

Form a user's perspective it's the time the background thread stays around to speed up token-refreshes.
It's rather the time the Nessie client will proactively refresh tokens.

Actually, the name of the parameter should also be something around that meaning and not "keep alive" - maybe "idle client token refresh timeout" or so (also not a brilliant idea...).

Should mention that setting this to a "too high" value _could_ is not a good way (thread leak, memory leak - if clients are re-created).

Reply:
`CONF_NESSIE_OAUTH2_IDLE_INTERVAL` -> `CONF_NESSIE_OAUTH2_PREEMPTIVE_TOKEN_REFRESH_IDLE_TIMEOUT` ? The "idle" part is a hint that the preemptive refresh will not happen when the client is idle.

`CONF_NESSIE_OAUTH2_BACKGROUND_THREAD_IDLE_TIMEOUT` - LGTM.

Reply:
Also, I think we can keep the constant names shorter as they are not user-visible, but have descriptive property names (values of those constants).